esp/services/ws_sql/ws_sqlService.cpp

```diff
@@ -716,8 +715,7 @@ bool CwssqlEx::getWUResult(IEspContext &context, const char * wuid, StringBuffer
 
 bool CwssqlEx::onSetRelatedIndexes(IEspContext &context, IEspSetRelatedIndexesRequest &req, IEspSetRelatedIndexesResponse &resp)
 {
-    if (!context.validateFeatureAccess(WSSQLACCESS, SecAccess_Write, false))
-        throw MakeStringException(-1, "WsSQL::SetRelatedIndexes failed to execute SQL. Permission denied.");
+    context.ensureFeatureAccess(WSSQLACCESS, SecAccess_Write, -1, "WsSQL::SetRelatedIndexes failed to execute SQL. Permission denied.");
```

Comment:
original error message was wrong. should be "WsSQL::SetRelatedIndexes: Permission denied."